### DISCUSSION TITLE
Exclude database from reports

### DISCUSSION THREAD

**Discussion by Phnorma** [zubkov-andrei/pg_profile#50] [General]
Добрый день!
В кластере 2 бд, но отчет необходимо получить только по одной.
Пробовал фильтровать не нужную бд через set_server_db_exclude(server name, exclude_db name[]), но даже после этого продолжаю получать данные по исключенной базе данных.
Подскажите как мне лучше реализовать фильтрацию?

**Comment by zubkov-andrei** [comment]
Добрый день!
Уточните, пожалуйста, а какую цель вы преследуете?
Полноценной фильтрации баз данных в pg_profile пока нет. Функция _set_server_db_exclude()_ предназначена для того, чтобы не подключаться к базам, к которым подключение невозможно (Например в AWS). В то же время pg_stat_statements, например, показывает данные по выражениям, исполняемым во всех базах.
Сама же идея исключения информации о конкретных базах выглядит странно из-за того, что pg_profile по задумке наблюдает за работой кластера а целом. Если, например, окажется что основная нагрузка на кластер будет в исключенной базе - вы увидите что в отчете нагрузки существенной не видно, а всё тормозит. Причину так не найти.

**Reply by Phnorma** [reply]
В кластере есть 2 базы и при активной работе одна забивает другую в статистике, получается только pg_profile.topn увеличивать, в надежде что в топ запросов попадут запросу и из второй бд?

**Reply by zubkov-andrei** [reply]
Это понятно, но зачем вам поиск мелочей? Заслуживает ли внимания нагрузка в ненагруженной базе, если кластер занят в основном другой работой?

**Reply by Phnorma** [reply]
Я видимо не совсем корректно выразился.
Нагрузка есть в обоих бд. Это абсолютно разные по смыслу бд, поэтому хотелось бы понимать как выполняются запросы не только в той что генерирует бОльшую нагрузку.
Тут получается или выносить в другой кластер или как я выше писал расширять список.
В любом случае, спасибо за отличное расширение и ответ.
